src/main/java/com/hashedin/bitly/UrlShortenerController.java

```diff
@@ -0,0 +1,32 @@
+package com.hashedin.bitly;
+
+import java.io.IOException;
+import java.net.URI;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@Controller
+public class UrlShortenerController {
+
+	private final ShortCodeMappingService service;
+	
+	@Autowired
+	public UrlShortenerController(ShortCodeMappingService service) {
+		this.service = service;
+	}
```

Comment:
We don't autowire the member variable. Instead, we autowire the consutructor. This allows our member variables to be final, and makes our controller immutable.